app/scripts/components/common/cookie-consent/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState, useEffect, useMemo } from 'react';
+
+import { Icon } from '@trussworks/react-uswds';
+import {
+  USWDSAlert,
+  USWDSButton,
+  USWDSButtonGroup,
+  USWDSLink
+} from '$components/common/uswds';
+
+import './index.scss';
+
+interface CookieConsentProps {
+  title?: string | undefined;
+  copy?: string | undefined;
+  onFormInteraction: () => void;
+}
+
+export const CookieConsent = ({
+  title,
+  copy,
+  onFormInteraction
+}: CookieConsentProps) => {
+  const [cookieConsentResponded, SetCookieConsentResponded] =
```

Comment:
I would like to learn more about where to save cookie consent prefrence. We have been saving the user preference on local storage so far - can you leave some notes about why we are using cookie for this case? Moreover, when user says no to cookie usage, can we save this info to cookie at all?

Reply:
We can change this to local storage if that is the existing pattern across the system. 

Based on what I could find through my original research, the prevailing pattern is to store the consent in a cookie since they require an expiration associated with the cookies. Allowing for redundancy when a user clears their cache and cookies they will be re-prompted with cookie consent interface. Because of the temporary nature it appears to be the primary storage method (while contradictory to the implied practice).